packages/react-dom/src/client/ReactDOM.js

```diff
@@ -756,7 +757,11 @@ const DOMRenderer = ReactFiberReconciler({
     ): void {
       if (container.nodeType === COMMENT_NODE) {
         (container.parentNode: any).insertBefore(child, container);
+        trapClickOnNonInteractiveElement(
+          ((container.parentNode: any): HTMLElement),
+        );
       } else {
+        trapClickOnNonInteractiveElement(((container: any): HTMLElement));
```

Comment:
What if it already has `onclick`? Is it expected that rendering into it overrides `onclick`?

Reply:
I guess if it already has `onclick` we don't need to trap click events, so I'll wrap this in a null check.

Reply:
Should we do the same in `trapClickOnNonInteractiveElement` itself then?

Reply:
Yeah that's what I was doing :) my only hesitation is that `trapClickOnNonInteractiveElement` is also called on elements that React controls (which shouldn't have `onclick` making the check redundant in most cases).

Reply:
Yeah, I guess you're right

Reply:
By the way do we do this regardless of whether we're mobile safari? Is that good? Should we check the browser somehow?

Reply:
Right now we do it regardless of the browser. There's an [existing TODO for adding browser detection](https://github.com/facebook/react/blob/master/packages/react-dom/src/client/ReactDOMFiberComponent.js#L263). It might be a good time to address that, assuming browser detection is fast/reliable enough.